src/libraries/System.ComponentModel.Primitives/src/System/ComponentModel/Design/Serialization/DesignerSerializerAttribute.cs

```diff
@@ -27,8 +27,8 @@ public DesignerSerializerAttribute(Type serializerType, Type baseSerializerType)
                 throw new ArgumentNullException(nameof(baseSerializerType));
             }
 
-            SerializerTypeName = serializerType.AssemblyQualifiedName;
-            SerializerBaseTypeName = baseSerializerType.AssemblyQualifiedName;
+            SerializerTypeName = serializerType.AssemblyQualifiedName!;
```

Comment:
There's nothing guaranteeing that `SerializerTypeName` and `SerializerBaseTypeName` aren't null. Would it make more sense to annotate these as `string?`? I see some code inspecting them for null.

[https://github.com/dotnet/winforms/blob/6163db0c3740bba5070e1a633a8d3e159a0fc23b/src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/DesignerSerializationManager.cs#L432](https://github.com/dotnet/winforms/blob/6163db0c3740bba5070e1a633a8d3e159a0fc23b/src/System.Windows.Forms.Design/src/System/ComponentModel/Design/Serialization/DesignerSerializationManager.cs#L432)

Reply:
Sounds good.